internal/appsec/emitter/sharedsec/shared.go

```diff
@@ -39,7 +40,7 @@ var userIDOperationArgsType = reflect.TypeOf((*UserIDOperationArgs)(nil)).Elem()
 func ExecuteUserIDOperation(parent dyngo.Operation, args UserIDOperationArgs) error {
 	var err error
 	op := &UserIDOperation{Operation: dyngo.NewOperation(parent)}
-	dyngo.OnData(op, func(e error) { err = e })
+	dyngo.OnData(op, func(e *events.SecurityBlockingEvent) { err = e })
 	dyngo.StartOperation(op, args)
```

Comment:
it's minor but really is a mistake not to have the OnData listener under the user id operation scope.
Same for all other SecurityBlockingEvent listeners of this PR. It's just that StartOperation doesn't have the right API everywhere. We should align them all with `setup` like it was done in httpsec IIRC.